android/titanium/src/java/org/appcelerator/titanium/TiBlob.java

```diff
@@ -786,14 +791,18 @@ public TiBlob imageAsResized(Number width, Number height)
 		} catch (OutOfMemoryError e) {
 			TiBlobLruCache.getInstance().evictAll();
 			Log.e(TAG, "Unable to resize the image. Not enough memory: " + e.getMessage(), e);
-			return null;
 		} catch (IllegalArgumentException e) {
 			Log.e(TAG, "Unable to resize the image. Illegal Argument: " + e.getMessage(), e);
-			return null;
 		} catch (Throwable t) {
 			Log.e(TAG, "Unable to resize the image. Unknown exception: " + t.getMessage(), t);
-			return null;
+		} finally {
+			// Perform soft garbage collection to reclaim memory.
+			KrollRuntime instance = KrollRuntime.getInstance();
+			if (instance != null) {
```

Comment:
All these null checks remind me of the null object pattern. I know it's super common to check for null in Java and we have null checks everywhere, but does KrollRuntime.getInstance() ever realistically return null? If so, maybe have it return a dummy NullRuntime that has no-ops for things like `softGC()`? If not, mark the method as @NonNull and avoid the checks? Looks to me the only way it could be null is if somehow it got called before `init()` (though certainly it looks like there's som minimal code paths we hit before that happens, like the asset helper/cache stuff.

Reply:
To be honest, if `getInstance()` were `null` then something bad happened. But technically, it is possible for it to be `null`.

I'm going to set it to `@Nullable` and leave a TODO for future reference so we can clean up our codebase.